docs/docs/dev_docs/contracts/syntax/storage.md

```diff
@@ -1,40 +1,533 @@
+
+#include_code map /yarn-project/aztec-nr/aztec/src/state_vars/map.nr rust
+
+It includes a [`Context`](./context.mdx) to be used when dabbling in the private domain, a `storage_lot` to specify where in storage the map is stored, and a `start_var_constructor` which tells the map how it should operate on the underlying type. This includes how to serialize and deserialize the type, as well as how commitments and nullifiers are computed for the type if private.
+
+### `new`
+
+When declaring the storage for a map, we use the `Map::new()` constructor. As seen below, this takes the `storage_slot` and the `start_var_constructor` along with the [`Context`](./context.mdx).
+
+#include_code new /yarn-project/aztec-nr/aztec/src/state_vars/map.nr rust
+
+We will see examples of map constructors for public and private variables in later sections.
+
+### `at`
+
+When dealing with a map, we can access the value at a given key using the `::at` method. This takes the key as an argument and returns the value at that key.
+
+#include_code at /yarn-project/aztec-nr/aztec/src/state_vars/map.nr rust
+
+This function behaves similarly for both private and public maps. An example could be if we have a map with `minters`, which is mapping addresses to a flag for whether they are allowed to mint tokens or not.
+
+#include_code read_minter /yarn-project/noir-contracts/src/contracts/token_contract/src/main.nr rust
+
+Above, we are specifying that we want to get the storage in the Map `at` the `msg_sender()`, read the value stored and check that `msg_sender()` is indeed a minter. Doing a similar operation in Solidity code would look like:
+
+```solidity
+require(minters[msg.sender] == 1, "caller is not minter");
+```
+
+## Public State Variables
+
+As we have looked seen earlier we got two kinds of state, public and private. We'll start with a look at the public side of things, as this is what behaves closest to what most people are used to from Ethereum and other blockchains.
+
+To define that a variable is public, it is wrapped in the `PublicState` struct, as defined below:
+
+#include_code public_state_struct /yarn-project/aztec-nr/aztec/src/state_vars/public_state.nr rust
+
+The `PublicState` struct is generic over the variable type `T` and its serialized size `T_SERIALISED_LEN`. 
+:::info
+Currently, the length of the types must be specified when declaring the storage struct but the intention is that this will be inferred in the future.
 :::
 
-In [State Variables](./state_variables.md) we will see in more detail what each of these types are, how they work and how to initialize them.
+The struct contains a `storage_slot` which similarly to Ethereum is used to figure out *where* in the storage the variable is stored. Notice that while we don't have the exact same [state model](./../../../concepts/foundation/state_model.md) as EVM chains it will from the developers points of view work very similar.
```

Comment:
```suggestion
The struct contains a `storage_slot` which, similar to Ethereum, is used to figure out *where* in storage the variable is located. Notice that while we don't have the exact same [state model](./../../../concepts/foundation/state_model.md) as EVM chains it will look similar from the contract developers point of view.
```